cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggeringInfoHelper.java

```diff
@@ -0,0 +1,181 @@
+      default:
+        return null;
+    }
+  }
+
+  private static ProgramRunId getProgramRunId(DefaultProgramStatusTriggerInfo programStatusTriggerInfo) {
+    return new ProgramRunId(programStatusTriggerInfo.getNamespace(), programStatusTriggerInfo.getApplicationName(),
+                            programTypeMappings.get(programStatusTriggerInfo.getProgramType()),
+                            programStatusTriggerInfo.getProgram(), programStatusTriggerInfo.getRunId().getId());
+  }
+
+  private static TriggeringPropertyMapping triggeringPropertyMapping(Map<String, String> properties) {
+    if (!properties.containsKey(TRIGGERING_PROPERTIES_MAPPING)) {
+      return null;
+    }
+    return gson.fromJson(properties.get(TRIGGERING_PROPERTIES_MAPPING), TriggeringPropertyMapping.class);
+  }
+
+  /**
+   * ProgramType mappings for {@link ProgramType} and {@link io.cdap.cdap.api.app.ProgramType}
+   */
+  static Map<io.cdap.cdap.api.app.ProgramType, ProgramType> programTypeMappings = new HashMap<>();
+
+  static {
+    programTypeMappings.put(io.cdap.cdap.api.app.ProgramType.WORKFLOW, ProgramType.WORKFLOW);
+    programTypeMappings.put(io.cdap.cdap.api.app.ProgramType.MAPREDUCE, ProgramType.MAPREDUCE);
+    programTypeMappings.put(io.cdap.cdap.api.app.ProgramType.SERVICE, ProgramType.SERVICE);
+    programTypeMappings.put(io.cdap.cdap.api.app.ProgramType.SPARK, ProgramType.SPARK);
+    programTypeMappings.put(io.cdap.cdap.api.app.ProgramType.WORKER, ProgramType.WORKER);
+  }
+
+  private static List<TriggeringInfo> getTriggeringInfoList(List<TriggerInfo> triggerInfos, ScheduleId scheduleId) {
+    return triggerInfos.stream().map(r -> fromTriggerInfo(r, scheduleId)).collect(Collectors.toList());
+  }
+
+  private static TriggeringInfo fromTriggerInfo(TriggerInfo triggerInfo, ScheduleId scheduleId) {
+    switch(triggerInfo.getType()) {
+      case PROGRAM_STATUS:
+        DefaultProgramStatusTriggerInfo progTrigger = (DefaultProgramStatusTriggerInfo) triggerInfo;
+        return new TriggeringInfo.ProgramStatusTriggeringInfo(
+          scheduleId, progTrigger.getRuntimeArguments(), getProgramRunId(progTrigger));
+      case TIME:
+        DefaultTimeTriggerInfo timeTrigger = (DefaultTimeTriggerInfo) triggerInfo;
+        return new TriggeringInfo.TimeTriggeringInfo(
+          scheduleId, null, timeTrigger.getCronExpression());
+      case PARTITION:
+        DefaultPartitionTriggerInfo partTrigger = (DefaultPartitionTriggerInfo) triggerInfo;
+        return new TriggeringInfo.PartitionTriggeringInfo(
+          scheduleId, null, partTrigger.getDatasetName(), partTrigger.getDatasetNamespace(),
+          partTrigger.getExpectedNumPartitions(), partTrigger.getActualNumPartitions());
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Serializer/Deserializer for {@link TriggeringInfo}
+   */
+  public static class TriggeringInfoCodec implements JsonDeserializer<TriggeringInfo>,
```

Comment:
nit: Can this be an independent class since the usage is not tied to this class?